contracts/governance/StakingRewards/StakingRewardsOs.sol

```diff
@@ -0,0 +1,256 @@
+        );
+        checkpointBlockNumber[_checkpointTime] = checkpointBlock;
+    }
+
+    /**
+     * @notice Get staker's current accumulated reward
+     * @dev The getStakerCurrentReward() function internally calls this function to calculate reward amount
+     * @param _considerMaxDuration True: Runs for the maximum duration - used in tx not to run out of gas
+     * False - to query total rewards
+     * @param _startTime The time from which the staking rewards calculation shall restart.
+     * @return The timestamp of last withdrawal
+     * @return The accumulated reward
+     */
+    function getStakerCurrentReward(
+        bool _considerMaxDuration,
+        uint256 _startTime
+    ) external view returns (uint256 nextWithdrawTimestamp, uint256 amount) {
+        return _getStakerCurrentReward(msg.sender, _considerMaxDuration, _startTime);
+    }
+
+    /**
+     * @notice Internal function to calculate staker's current reward
+     * @dev Normally the start time is 0. If this function returns a valid withdraw timestamp
+     * and zero amount - that means there were no valid rewards for that period. So the new period must start
+     * from the end of the last interval or till the time no rewards are accumulated i.e. _startTime
+     */
+    function _getStakerCurrentReward(
+        address _staker,
+        bool _considerMaxDuration,
+        uint256 _startTime
+    ) internal view returns (uint256 nextWithdrawTimestamp, uint256 amount) {
+        uint256 weightedStake;
+        uint256 lastFinalisedBlock;
+        uint256 startWithdrawTimestamp;
+        uint256 endWithdrawTimestamp;
+
+        startWithdrawTimestamp = stakerNextWithdrawTimestamp[_staker];
+        if (stopBlock != 0 && startWithdrawTimestamp >= stopRewardsTimestamp) {
+            return (startWithdrawTimestamp, 0);
+        }
+
+        uint256 currentBlockTsToLockDate = staking.timestampToLockDate(block.timestamp);
+
+        if (startWithdrawTimestamp == 0) {
+            startWithdrawTimestamp = rewardsProgramStartTime;
+        }
+        if (currentBlockTsToLockDate <= startWithdrawTimestamp) {
+            return (startWithdrawTimestamp, 0);
+        }
+        if (startWithdrawTimestamp <= _startTime) {
+            startWithdrawTimestamp = staking.timestampToLockDate(_startTime);
+        }
+
+        if (_considerMaxDuration) {
+            endWithdrawTimestamp = startWithdrawTimestamp.add(maxDuration);
+            endWithdrawTimestamp = endWithdrawTimestamp < block.timestamp
+                ? staking.timestampToLockDate(endWithdrawTimestamp)
+                : currentBlockTsToLockDate;
+        } else {
+            endWithdrawTimestamp = currentBlockTsToLockDate;
+        }
+
+        if (stopBlock != 0 && endWithdrawTimestamp > stopRewardsTimestamp) {
+            // add 2 weeks because the right boundary is not included in processing cycle below
+            endWithdrawTimestamp = stopRewardsTimestamp.add(TWO_WEEKS);
+        }
```

Comment:
i considered both comments, but kept stop block check here and removed from _computeWeightedStakeForDate

Reply:
should not be an issue with the old code because this is ony important when the program actually stops.  
but i will create a ticket to check this assumption.